sdk/core/Azure.Core/src/Pipeline/RetryPolicy.cs

```diff
@@ -34,6 +36,18 @@ public RetryPolicy(int maxRetries = 3, DelayStrategy? delayStrategy = default)
             _delayStrategy = delayStrategy ?? DelayStrategy.CreateExponentialDelayStrategy();
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RetryPolicy"/> class.
+        /// </summary>
+        /// <param name="options">The set of options to use for configuring the retry policy.</param>
+        public RetryPolicy(RetryPolicyOptions options)
+        {
+            Argument.AssertNotNull(options, nameof(options));
+            _maxRetries = options.MaxRetries;
+            _delayStrategy = options.DelayStrategy ?? DelayStrategy.CreateExponentialDelayStrategy();
+            _responseClassifier = options.ResponseClassifier;
```

Comment:
Would it make sense to take the one from the pipeline as a default if this isn't specified in options?

Reply:
We don't have access to the pipeline at this point.